python/tank/platform/bundle.py

```diff
@@ -920,6 +935,37 @@ def _get_engine_name(self):
 
         return engine_name
 
+    def _get_metrics_context(self):
+        """
+        Should be re-implemented in deriving classes and return a dictionary with
+        the properties needed to log a metric event for this bundle.
+
+        :raises: NotImplementedError
+        """
+        raise NotImplementedError
+
+    def log_metric(self, action, log_version=False, log_once=False, command_name=None):
+        """
+        Log metrics for this bundle and the given action.
+
```

Comment:
please include examples for how to use this method and what it means to include the various parameters! This will help users trying to implement metrics. What does "a toolkit command name to add to metric properties" exactly - it's not clear. an example helps explain it to the user.

Reply:
I believe this method was adde for retro-compatibility, I feel we shoudl just remove the comment and clearly state it shouldn't be used.

Reply:
unless i am mistaken, this is the main API method for analytics